### DISCUSSION TITLE
about multi teachable agent

### DISCUSSION THREAD

**Discussion by zhuxm-ai** [microsoft/autogen#3666] [Q&A]
The teachable agent is amazing. I have two questions:
1.When there are multiple teachable agents, do we need multiple Teachability instances?
2.When teaching the agent many times, does the content of the next teaching session overwrite the content of the previous teaching session? 
thx for answer

**Comment by rickyloynd-microsoft** [comment]
Glad you like teachable agents!
1. Yes. Each instance of the Teachability class retains one agent's long-term memory.
2. No, the memories are cumulative.
